executors/cpp/datetime_fmt.cpp

```diff
@@ -0,0 +1,257 @@
+/********************************************************************
+ * Comments and license as needed
+ ************************************
+
+/******
+ * testing datetime format
+ */
+
+
+#include "unicode/utypes.h"
+#include "unicode/unistr.h"
+#include "unicode/datefmt.h"
+#include "unicode/locid.h"
+#include "unicode/smpdtfmt.h"
+#include "unicode/uclean.h"
+
+#include "util.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <json-c/json.h>
+
+#include <iostream>
+#include <string>
+#include <cstring>
+
+using icu::DateFormat;
+
+using std::cout;
+using std::endl;
+using std::string;
+
+icu::DateFormat::EStyle stringToEStyle(string style_string) {
+  if (style_string == "full") return icu::DateFormat::kFull;
+  if (style_string == "long") return icu::DateFormat::kLong;
+  if (style_string == "medium") return icu::DateFormat::kMedium;
+  if (style_string == "short") return icu::DateFormat::kShort;
+  return icu::DateFormat::kDefault;
+}
+
+
+const string test_datetime_fmt(json_object *json_in) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  json_object *label_obj = json_object_object_get(json_in, "label");
+  string label_string = json_object_get_string(label_obj);
+
+  // The locale for formatted output
+  json_object *locale_label_obj = json_object_object_get(json_in, "locale");
+  string locale_string;
+  if (locale_label_obj) {
+    locale_string = json_object_get_string(locale_label_obj);
+  } else {
+    locale_string = "und";
+  }
+
+  Locale displayLocale(locale_string.c_str());
+
+  DateFormat* df;
+
+  // Get the input data as a date object.
+
+  // Types of input:
+  //   "input_millis" milliseconds since base date/time
+  //   "input_string" parsable string such as "2020-03-02 10:15:17 -08:00"
+  //   OTHER?
+
+  string dateStyle_str;
+  string timeStyle_str;
+  string calendar_str;
+  string timezone_str;
+
+  string era_str;
+  string year_str;
+  string month_str;
```

Comment:
Question: I don't see where you read these variables. Intentional?

Reply:
They will be used when I figure out the API to handle individual fields in CPP version of DateTime format.